client/interfaces.go

```diff
@@ -26,30 +26,44 @@ import (
 
 // Plug represents a capacity offered by a snap.
 type Plug struct {
+	Snap string `json:"snap"`
 	// NOTE: the discrepancy between go and json field name is intentional.
-	Name      string                 `json:"plug"`
-	Snap      string                 `json:"snap"`
-	Interface string                 `json:"interface,omitempty"`
-	Attrs     map[string]interface{} `json:"attrs,omitempty"`
-	Apps      []string               `json:"apps,omitempty"`
-	Label     string                 `json:"label,omitempty"`
+	Name        string                 `json:"plug"`
+	Interface   string                 `json:"interface,omitempty"`
+	Attrs       map[string]interface{} `json:"attrs,omitempty"`
+	Apps        []string               `json:"apps,omitempty"`
+	Label       string                 `json:"label,omitempty"`
+	Connections []SlotRef              `json:"connections,omitempty"`
+}
+
+// PlugRef is a reference to a plug.
+type PlugRef struct {
+	Snap string `json:"snap"`
+	Name string `json:"plug"`
 }
 
 // Slot represents the potential of a given snap to connect to a given plug.
 type Slot struct {
+	Snap string `json:"snap"`
 	// NOTE: the discrepancy between go and json field name is intentional.
-	Name      string                 `json:"slot"`
-	Snap      string                 `json:"snap"`
-	Interface string                 `json:"interface,omitempty"`
-	Attrs     map[string]interface{} `json:"attrs,omitempty"`
-	Apps      []string               `json:"apps,omitempty"`
-	Label     string                 `json:"label,omitempty"`
+	Name        string                 `json:"slot"`
+	Interface   string                 `json:"interface,omitempty"`
+	Attrs       map[string]interface{} `json:"attrs,omitempty"`
+	Apps        []string               `json:"apps,omitempty"`
+	Label       string                 `json:"label,omitempty"`
+	Connections []PlugRef              `json:"connections,omitempty"`
+}
+
+// SlotRef is a reference to a slot.
+type SlotRef struct {
+	Snap string `json:"snap"`
+	Name string `json:"slot"`
 }
 
-// PlugConnections represents a single plug and slots that are connected to it.
-type PlugConnections struct {
-	Plug
-	Connections []Slot `json:"connections"`
+// InterfaceConnections contains information about all plugs, slots and their connections
+type InterfaceConnections struct {
```

Comment:
s/InterfaceConnections/Interfaces/

This lists plugs and slots, whether connected or not.